feat(export): Add D2 class diagram output format

Else00 opened the pull request:
### Summary

This PR introduces a new output format to `erdantic`, allowing users to generate class diagrams using the [D2 declarative diagramming language](https://d2lang.com/).

### Motivation

While `erdantic`'s default Graphviz output is excellent, D2 offers powerful, modern layout engines (like ELK and Dagre) that can be particularly effective for organizing large, complex data models automatically. This provides users with an alternative rendering backend that excels at creating clean, readable diagrams from complex schemas.

As a text-based format, D2 diagrams are also highly portable and easy to version control alongside source code.

For more information on D2's class diagrams, see the official tour: https://d2lang.com/tour/uml-classes

### Implementation

- A new `to_d2()` method has been added to the `EntityRelationshipDiagram` class.
- A corresponding `--d2` / `-D` CLI flag has been added to generate D2 output to stdout.
- The implementation maps `erdantic`'s model and relationship metadata to D2's UML class shape syntax, including field visibility based on Python naming conventions (`_` for protected, `__` for private).

### Usage and Example Output

The commands below generate a `diagram.d2` file from the `pydantic` models and then render it into an SVG:

```bash
# Generate D2 output for the pydantic example models
erdantic --d2 erdantic.examples.pydantic.Party > diagram.d2

# Render the diagram to SVG using the D2 CLI
d2 diagram.d2 diagram.svg
```

Here is the resulting `diagram.svg`:

![diagram](https://github.com/user-attachments/assets/a90ab947-9657-4099-9b32-d72e25914a7e)

<br>

<details>
<summary><b>Layout Engine Comparison on a Complex Diagram</b></summary>

To showcase the power of D2's layout engines, here are renderings of a more complex diagram using `dagre`, `elk`, and `tala`.

**1. Dagre (Default Layout)**
`d2 -l dagre complex.d2 d2_dagre.svg`

![d2_dagre](https://github.com/user-attachments/assets/53fcaca8-b4a3-4aca-8159-fcb166e8ea56)

---

**2. ELK Layout**
`d2 -l elk complex.d2 d2_elk.svg`

![d2_elk](https://github.com/user-attachments/assets/2a751c68-c072-4a32-86d4-ac01220b87bd)

---

**3. TALA Layout**
`d2 -l tala complex.d2 d2_tala.svg`

![d2_tala](https://github.com/user-attachments/assets/fc2e67f5-4afb-46e0-a13d-38f2c41d47f8)

</details>

jayqi commented:
Hi @Else00,

Thanks for the very polished PR! This functionality looks great, and I'm definitely interested in getting it into erdantic. Just as a heads up—I'm pretty busy with work and travel over the next two weeks. Since adding support for a new output format and extending the API is a big deal, I want to be thoughtful and make sure the details of the proposed implementation make sense. D2 is also new to me, so I'll want to learn at least a little more about it. That's just all to say that I may be slow to review, but that doesn't mean I'm abandoning this or don't want to merge. If you don't get a review from me by July 16, please ping me here.

jayqi commented:
Please note that I pushed some commits that you need to pull.

Else00 commented:
Implemented the requested changes:

- Removed the `-D` short alias; kept only `--d2`.
- Switched to D2 crow’s-foot arrowheads (`cf-one`, `cf-one-required`, `cf-many`, `cf-many-required`), handling `UNSPECIFIED` combinations as discussed.
- Always quote identifiers for D2 (shape keys and connection labels). Do not quote field names in class bodies to match snapshots.
- Quote type values only when needed (special chars), now including the `|` for PEP 604 unions.
- Kept the lazy import of the D2 renderer inside `to_d2()` and documented it to avoid circular imports.
- Updated tests and D2 asset snapshots accordingly. Local test suite: 118 passed.

Happy to tweak mapping defaults or quoting strategy if you prefer something different.

jayqi commented:
Hi @Else00, sorry for the delay. I will take a look at this PR over the weekend.
